Unable to download tile from local repository

I have built a tile and stored it in our local Nexus repository.  If the tile is in my .m2 directory, I can use it.  However, if I delete it from there and point a pom.xml at the local Nexus, I am unable to download it.

How do I get it to download a tile that it not published to Maven Central?

`pom.xml`

```

<project>
    <modelVersion>4.0.0</modelVersion>

    <groupId>io.repaint.maven</groupId>
    <artifactId>license-tile</artifactId>
    <version>1.1-SNAPSHOT</version>
    <packaging>tile</packaging>
    <description>Contains consistent license information.</description>

    <properties>
        <external.repository.url> path to local nexus repository</external.repository.url>
    </properties>

    <repositories>
        <repository>
            <id>my-repo</id>
            <name>Local Nexus</name>
            <url>${external.repository.url}</url>
        </repository>
        <repository>
            <id>central</id>
            <name>Local Nexus</name>
            <url>${external.repository.url}</url>
        </repository>
    </repositories>

    <pluginRepositories>
        <pluginRepository>
            <snapshots>
                <enabled>false</enabled>
            </snapshots>
            <id>central</id>
            <name>repo</name>
            <url>${external.repository.url}</url>
        </pluginRepository>
    </pluginRepositories>

    <build>
        <plugins>
            <plugin>
                <groupId>io.repaint.maven</groupId>
                <artifactId>tiles-maven-plugin</artifactId>
                <version>2.12</version>
                <extensions>true</extensions>
                <configuration>
                    <tiles>
                       <!-- Both versions work if the tile is in ~/.m2.  Neither works if it is only on Nexus -->
                        <!-- <tile>com.str.tiles:buildJavaPlay:[1.0.0,2.0.0)</tile> -->
                        <tile>com.str.tiles:buildJavaPlay:1.0.2</tile>
                    </tiles>
                </configuration>
            </plugin>
        </plugins>
    </build>
</project>
```

Output:

```
$ mvn -U install
[INFO] Scanning for projects...
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/Users/me/.m2/repository/org/codehaus/groovy/groovy-all/2.4.14/groovy-all-2.4.14.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Downloading from central: https://repo.maven.apache.org/maven2/com/str/tiles/buildJavaPlay/1.0.2/buildJavaPlay-1.0.2.xml
[ERROR] Could not find artifact com.str.tiles:buildJavaPlay:xml:1.0.2 in central (https://repo.maven.apache.org/maven2)
[ERROR] 
[ERROR] Try downloading the file manually from the project website.
[ERROR] 
[ERROR] Then, install it using the command: 
[ERROR]     mvn install:install-file -DgroupId=com.str.tiles -DartifactId=buildJavaPlay -Dversion=1.0.2 -Dpackaging=xml -Dfile=/path/to/file
[ERROR] 
[ERROR] Alternatively, if you host your own repository you can deploy the file there: 
[ERROR]     mvn deploy:deploy-file -DgroupId=com.str.tiles -DartifactId=buildJavaPlay -Dversion=1.0.2 -Dpackaging=xml -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR] 
[ERROR] 
[ERROR]   com.str.tiles:buildJavaPlay:xml:1.0.2
[ERROR] 
[ERROR] from the specified remote repositories:
[ERROR]   central (https://repo.maven.apache.org/maven2, releases=true, snapshots=false)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException

Compilation exited abnormally with code 1 at Fri Jan  4 15:40:32
```

Why is it trying to download it from Central? You can see in the messages
that is where it is trying to download it from. Do you not have your Nexus
setup as a Release repository?

On Sat, Jan 5, 2019 at 9:50 AM udalrich <notifications@github.com> wrote:

> I have built a tile and stored it in our local Nexus repository. If the
> tile is in my .m2 directory, I can use it. However, if I delete it from
> there and point a pom.xml at the local Nexus, I am unable to download it.
>
> How do I get it to download a tile that it not published to Maven Central?
>
> pom.xml
>
>
> <project>
>     <modelVersion>4.0.0</modelVersion>
>
>     <groupId>io.repaint.maven</groupId>
>     <artifactId>license-tile</artifactId>
>     <version>1.1-SNAPSHOT</version>
>     <packaging>tile</packaging>
>     <description>Contains consistent license information.</description>
>
>     <properties>
>         <external.repository.url> path to local nexus repository</external.repository.url>
>     </properties>
>
>     <repositories>
>         <repository>
>             <id>my-repo</id>
>             <name>Local Nexus</name>
>             <url>${external.repository.url}</url>
>         </repository>
>         <repository>
>             <id>central</id>
>             <name>Local Nexus</name>
>             <url>${external.repository.url}</url>
>         </repository>
>     </repositories>
>
>     <pluginRepositories>
>         <pluginRepository>
>             <snapshots>
>                 <enabled>false</enabled>
>             </snapshots>
>             <id>central</id>
>             <name>repo</name>
>             <url>${external.repository.url}</url>
>         </pluginRepository>
>     </pluginRepositories>
>
>     <build>
>         <plugins>
>             <plugin>
>                 <groupId>io.repaint.maven</groupId>
>                 <artifactId>tiles-maven-plugin</artifactId>
>                 <version>2.12</version>
>                 <extensions>true</extensions>
>                 <configuration>
>                     <tiles>
>                        <!-- Both versions work if the tile is in ~/.m2.  Neither works if it is only on Nexus -->
>                         <!-- <tile>com.str.tiles:buildJavaPlay:[1.0.0,2.0.0)</tile> -->
>                         <tile>com.str.tiles:buildJavaPlay:1.0.2</tile>
>                     </tiles>
>                 </configuration>
>             </plugin>
>         </plugins>
>     </build>
> </project>
>
> Output:
>
> $ mvn -U install
> [INFO] Scanning for projects...
> WARNING: An illegal reflective access operation has occurred
> WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/Users/me/.m2/repository/org/codehaus/groovy/groovy-all/2.4.14/groovy-all-2.4.14.jar) to method java.lang.Object.finalize()
> WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
> WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
> WARNING: All illegal access operations will be denied in a future release
> Downloading from central: https://repo.maven.apache.org/maven2/com/str/tiles/buildJavaPlay/1.0.2/buildJavaPlay-1.0.2.xml
> [ERROR] Could not find artifact com.str.tiles:buildJavaPlay:xml:1.0.2 in central (https://repo.maven.apache.org/maven2)
> [ERROR]
> [ERROR] Try downloading the file manually from the project website.
> [ERROR]
> [ERROR] Then, install it using the command:
> [ERROR]     mvn install:install-file -DgroupId=com.str.tiles -DartifactId=buildJavaPlay -Dversion=1.0.2 -Dpackaging=xml -Dfile=/path/to/file
> [ERROR]
> [ERROR] Alternatively, if you host your own repository you can deploy the file there:
> [ERROR]     mvn deploy:deploy-file -DgroupId=com.str.tiles -DartifactId=buildJavaPlay -Dversion=1.0.2 -Dpackaging=xml -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
> [ERROR]
> [ERROR]
> [ERROR]   com.str.tiles:buildJavaPlay:xml:1.0.2
> [ERROR]
> [ERROR] from the specified remote repositories:
> [ERROR]   central (https://repo.maven.apache.org/maven2, releases=true, snapshots=false)
> [ERROR] -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
>
> Compilation exited abnormally with code 1 at Fri Jan  4 15:40:32
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/repaint-io/maven-tiles/issues/85>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADZ2PzkbSPSKxcobQFyl3WWPmP-IVTOks5u_76lgaJpZM4ZtOKA>
> .
>


-- 
---
Richard Vowles,
Full stack - from Kubernetes, through Node & Java, Web and Mobile
development in Flutter - software developer for hire!

ph: +64275467747, web: www.google.com/+RichardVowles

--- a/pom.xml
+++ b/pom.xml
@@ -82,11 +82,13 @@
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
       <version>${mavenVersion}</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.plugin-tools</groupId>
       <artifactId>maven-plugin-annotations</artifactId>
-      <version>3.4</version>
+      <version>3.3</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
@@ -97,7 +99,13 @@
     <dependency>
       <groupId>org.apache.maven.shared</groupId>
       <artifactId>maven-filtering</artifactId>
-      <version>1.3</version>
+      <version>3.1.1</version>
+      <exclusions>
+        <exclusion>
+          <groupId>org.apache.maven</groupId>
+          <artifactId>maven-core</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>com.bluetrainsoftware.composite</groupId>

--- a/src/it/frege-example/pom.xml
+++ b/src/it/frege-example/pom.xml
@@ -8,8 +8,9 @@
 	<name>Frege Tile Example</name>
 	<description>Ensure we have a plugin defined in the parent</description>
 	<url>http://repaint.io</url>
-
-
+	<properties>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+	</properties>
 	<build>
 		<plugins>
 			<plugin>

--- a/src/main/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipant.groovy
+++ b/src/main/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipant.groovy
@@ -27,12 +27,13 @@ import org.apache.maven.MavenExecutionException
 import org.apache.maven.artifact.Artifact
 import org.apache.maven.artifact.DefaultArtifact
 import org.apache.maven.artifact.handler.DefaultArtifactHandler
-import org.apache.maven.artifact.repository.ArtifactRepository
 import org.apache.maven.artifact.repository.ArtifactRepositoryFactory
 import org.apache.maven.artifact.repository.layout.ArtifactRepositoryLayout
 import org.apache.maven.artifact.resolver.ArtifactNotFoundException
 import org.apache.maven.artifact.resolver.ArtifactResolutionException
+import org.apache.maven.artifact.resolver.ArtifactResolutionRequest
 import org.apache.maven.artifact.resolver.ArtifactResolver
+import org.apache.maven.artifact.resolver.ResolutionErrorHandler
 import org.apache.maven.artifact.versioning.VersionRange
 import org.apache.maven.execution.MavenSession
 import org.apache.maven.model.Build
@@ -88,7 +89,6 @@ import static io.repaint.maven.tiles.GavUtil.getRealGroupId
 @Component(role = AbstractMavenLifecycleParticipant, hint = "TilesMavenLifecycleParticipant")
 public class TilesMavenLifecycleParticipant extends AbstractMavenLifecycleParticipant {
 
-	protected static final String TILE_EXTENSION = 'pom'
 	public static final TILEPLUGIN_GROUP = 'io.repaint.maven'
 	public static final TILEPLUGIN_ARTIFACT = 'tiles-maven-plugin'
 	public static final String TILEPLUGIN_KEY = "${TILEPLUGIN_GROUP}:${TILEPLUGIN_ARTIFACT}"
@@ -100,6 +100,9 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 	ArtifactResolver resolver
 
 	@Requirement
+	ResolutionErrorHandler resolutionErrorHandler
+
+	@Requirement
 	ModelBuilder modelBuilder
 
 	@Requirement
@@ -108,21 +111,13 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 	@Requirement
 	ProjectBuildingHelper projectBuildingHelper
 
-	/**
-	 * Component used to create a repository.
-	 */
-	ArtifactRepositoryFactory repositoryFactory;
+	@Requirement
+	ArtifactRepositoryFactory repositoryFactory
 
-	/**
-	 * Map that contains the layouts.
-	 */
 	@Requirement( role = ArtifactRepositoryLayout.class )
-	private Map<String, ArtifactRepositoryLayout> repositoryLayouts;
+	Map<String, ArtifactRepositoryLayout> repositoryLayouts
 
 	protected MavenVersionIsolator mavenVersionIsolate
-
-	List<ArtifactRepository> remoteRepositories
-	ArtifactRepository localRepository
 
 	NotDefaultModelCache modelCache
 
@@ -163,7 +158,7 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 		return getArtifactFromCoordinates(artifact.groupId, artifact.artifactId, 'pom', '', artifact.version)
 	}
 
-	protected Artifact resolveTile(MavenSession mavenSession, Artifact tileArtifact) throws MavenExecutionException {
+	protected Artifact resolveTile(MavenSession mavenSession, MavenProject project, Artifact tileArtifact) throws MavenExecutionException {
 		// try to find tile from reactor
 		if (mavenSession != null) {
 			List<MavenProject> allProjects = mavenSession.getProjects()
@@ -183,7 +178,11 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 			mavenVersionIsolate.resolveVersionRange(tileArtifact)
 
 			// Resolve the .xml file for the tile
-			resolver.resolve(tileArtifact, remoteRepositories, localRepository)
+			final def tileReq = new ArtifactResolutionRequest()
+				.setArtifact(tileArtifact)
+				.setRemoteRepositories(project?.remoteArtifactRepositories)
+				.setLocalRepository(mavenSession?.localRepository)
+			resolutionErrorHandler.throwErrors(tileReq, resolver.resolve(tileReq))
 
 			// When resolving from workspace (e.g. m2e) we might receive the path to pom.xml instead of the attached tile
 			if (tileArtifact.file && tileArtifact.file.name == "pom.xml") {
@@ -196,7 +195,11 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 
 			// Resolve the .pom file for the tile
 			Artifact pomArtifact = getPomArtifactForArtifact(tileArtifact)
-			resolver.resolve(pomArtifact, remoteRepositories, localRepository)
+			final def pomReq = new ArtifactResolutionRequest()
+				.setArtifact(pomArtifact)
+				.setRemoteRepositories(project?.remoteArtifactRepositories)
+				.setLocalRepository(mavenSession?.localRepository)
+			resolutionErrorHandler.throwErrors(pomReq, resolver.resolve(pomReq))
 
 			if (System.getProperty("performRelease")?.asBoolean()) {
 
@@ -283,15 +286,13 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 
 		this.mavenSession = mavenSession
 
-		this.remoteRepositories = mavenSession.request.remoteRepositories
-		this.localRepository = mavenSession.request.localRepository
-
 		this.mavenVersionIsolate = discoverMavenVersion(mavenSession)
 
 		this.modelCache = new NotDefaultModelCache(mavenSession)
 
-		repositoryFactory = mavenSession.container.lookup(ArtifactRepositoryFactory)
-		repositoryLayouts = mavenSession.lookupMap(ArtifactRepositoryLayout.class.getName()) as Map<String, ArtifactRepositoryLayout>
+		// disabled explicit lookup as these seem to be injected just fine. Are these required for eclipse m2e>
+		//repositoryFactory = mavenSession.container.lookup(ArtifactRepositoryFactory)
+		//repositoryLayouts = mavenSession.lookupMap(ArtifactRepositoryLayout.class.getName()) as Map<String, ArtifactRepositoryLayout>
 
 		List<MavenProject> allProjects = mavenSession.getProjects()
 		if (allProjects != null) {
@@ -366,7 +367,7 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 		parseConfiguration(project.model, project.getFile(), true)
 
 		// collect any unprocessed tiles, and process them causing them to potentially load more unprocessed ones
-		loadAllDiscoveredTiles(mavenSession)
+		loadAllDiscoveredTiles(mavenSession, project)
 
 		// don't do anything if there are no tiles
 		if (processedTiles) {
@@ -394,7 +395,7 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 		// parent structure
 		ModelSource2 mainArtifactModelSource = createModelSource(project.file)
 		ModelBuildingRequest request = new DefaultModelBuildingRequest(modelSource: mainArtifactModelSource,
-			pomFile: project.file, modelResolver: createModelResolver(), modelCache: modelCache,
+			pomFile: project.file, modelResolver: createModelResolver(project), modelCache: modelCache,
 		  systemProperties: mavenSession.request.systemProperties, userProperties: mavenSession.request.userProperties,
 			profiles: mavenSession.request.projectBuildingRequest.profiles,
 		  activeProfileIds: mavenSession.request.projectBuildingRequest.activeProfileIds,
@@ -482,6 +483,7 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 
 	ModelSource2 createModelSource(File pomFile) {
 		return new ModelSource2() {
+
 			InputStream stream = pomFile.newInputStream()
 
 			@Override
@@ -500,52 +502,56 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 			}
 
 			@Override
-			ModelSource2 getRelatedSource( String relPath ) {
-				File relatedPom = new File(pomFile.parentFile, relPath);
+			ModelSource2 getRelatedSource(String relPath) {
+				File relatedPom = new File(pomFile.parentFile, relPath)
 				if (relatedPom.isDirectory()) {
-					relatedPom = new File(relatedPom, "pom.xml");
+					relatedPom = new File(relatedPom, "pom.xml")
 				}
 				if (relatedPom.isFile()&& relatedPom.canRead()) {
-					return createModelSource(relatedPom.canonicalFile);
+					return createModelSource(relatedPom.canonicalFile)
 				}
-				return null;
+				return null
 			}
+
 		}
 	}
 
-	protected ModelResolver createModelResolver() {
+	protected ModelResolver createModelResolver(MavenProject project) {
 		// this is for resolving parents, so always poms
 
 		return new ModelResolver() {
+
+			@Override
 			ModelSource2 resolveModel(String groupId, String artifactId, String version) throws UnresolvableModelException {
                                 Artifact artifact = new DefaultArtifact(groupId, artifactId, VersionRange.createFromVersion(version), "compile",
 					"pom", null, new DefaultArtifactHandler("pom"))
 
 				mavenVersionIsolate.resolveVersionRange(artifact)
-				resolver.resolve(artifact, remoteRepositories, localRepository)
+				final def req = new ArtifactResolutionRequest()
+					.setArtifact(artifact)
+					.setRemoteRepositories(project?.remoteArtifactRepositories)
+					.setLocalRepository(mavenSession?.localRepository)
+				resolutionErrorHandler.throwErrors(req, resolver.resolve(req))
 
 				return createModelSource(artifact.file)
 			}
 
-			ModelSource2 resolveModel(Dependency dependency) throws UnresolvableModelException {
-				return resolveModel(dependency.groupId, dependency.artifactId, dependency.version)
-			}
-
+			@Override
 			ModelSource2 resolveModel(Parent parent) throws UnresolvableModelException {
 				return resolveModel(parent.groupId, parent.artifactId, parent.version)
 			}
 
+			@Override
 			void addRepository(Repository repository) throws InvalidRepositoryException {
 			}
 
+			@Override
 			void addRepository(Repository repository, boolean wat) throws InvalidRepositoryException {
 			}
 
-			void resetRepositories() {
-			}
-
+			@Override
 			ModelResolver newCopy() {
-				return createModelResolver()
+				return createModelResolver(project)
 			}
 		}
 
@@ -677,11 +683,11 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 		}
 	}
 
-	protected void loadAllDiscoveredTiles(MavenSession mavenSession) throws MavenExecutionException {
+	protected void loadAllDiscoveredTiles(MavenSession mavenSession, MavenProject project) throws MavenExecutionException {
 		while (unprocessedTiles.size() > 0) {
 			String unresolvedTile = unprocessedTiles.keySet().iterator().next()
 
-			Artifact resolvedTile = resolveTile(mavenSession, unprocessedTiles.remove(unresolvedTile))
+			Artifact resolvedTile = resolveTile(mavenSession, project, unprocessedTiles.remove(unresolvedTile))
 
 			TileModel tileModel = loadModel(resolvedTile)
 

--- a/src/test/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipantTest.groovy
+++ b/src/test/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipantTest.groovy
@@ -20,10 +20,11 @@ import groovy.transform.CompileStatic
 import io.repaint.maven.tiles.isolators.MavenVersionIsolator
 import org.apache.maven.MavenExecutionException
 import org.apache.maven.artifact.Artifact
-import org.apache.maven.artifact.repository.ArtifactRepository
-import org.apache.maven.artifact.resolver.ArtifactNotFoundException
 import org.apache.maven.artifact.resolver.ArtifactResolutionException
+import org.apache.maven.artifact.resolver.ArtifactResolutionRequest
+import org.apache.maven.artifact.resolver.ArtifactResolutionResult
 import org.apache.maven.artifact.resolver.ArtifactResolver
+import org.apache.maven.artifact.resolver.DefaultResolutionErrorHandler
 import org.apache.maven.execution.MavenSession
 import org.apache.maven.model.Build
 import org.apache.maven.model.Model
@@ -96,6 +97,12 @@ public class TilesMavenLifecycleParticipantTest {
 	void stuffParticipant() {
 		participant.logger = logger
 		participant.resolver = mockResolver
+		participant.resolver = [
+			resolve: { ArtifactResolutionRequest request ->
+				return new ArtifactResolutionResult()
+			}
+		] as ArtifactResolver
+		participant.resolutionErrorHandler = new DefaultResolutionErrorHandler()
 		participant.mavenVersionIsolate = createFakeIsolate()
 	}
 
@@ -132,7 +139,7 @@ public class TilesMavenLifecycleParticipantTest {
 		Artifact snapshot = getTileTestCoordinates()
 		System.setProperty(PERFORM_RELEASE, "true")
 		shouldFail(MavenExecutionException) {
-			participant.resolveTile(null, snapshot)
+			participant.resolveTile(null, null, snapshot)
 		}
 	}
 
@@ -141,22 +148,22 @@ public class TilesMavenLifecycleParticipantTest {
 		Artifact badbadbad = participant.getArtifactFromCoordinates("bad", "bad", "bad", "bad", "bad")
 
 		participant.resolver = [
-			resolve: { Artifact artifact, List<ArtifactRepository> remoteRepositories, ArtifactRepository localRepository ->
-				throw new ArtifactResolutionException("failed", badbadbad)
+			resolve: { ArtifactResolutionRequest request ->
+				new ArtifactResolutionResult().addErrorArtifactException(new ArtifactResolutionException("failed", badbadbad))
 			}
 		] as ArtifactResolver
 
 		shouldFail(MavenExecutionException) {
-			participant.resolveTile(null, badbadbad)
+			participant.resolveTile(null, null, badbadbad)
 		}
 		participant.resolver = [
-			resolve: { Artifact artifact, List<ArtifactRepository> remoteRepositories, ArtifactRepository localRepository ->
-				throw new ArtifactNotFoundException("failed", badbadbad)
+			resolve: { ArtifactResolutionRequest request ->
+				new ArtifactResolutionResult().addMissingArtifact(badbadbad)
 			}
 		] as ArtifactResolver
 
 		shouldFail(MavenExecutionException) {
-			participant.resolveTile(null, badbadbad)
+			participant.resolveTile(null, null, badbadbad)
 		}
 	}
 
@@ -240,7 +247,7 @@ public class TilesMavenLifecycleParticipantTest {
 			}
 		] as MavenVersionIsolator
 
-		def resolver = participant.createModelResolver()
+		def resolver = participant.createModelResolver(null)
 		def model = resolver.resolveModel('my', 'left', 'foot')
 
 		assert model.inputStream.text == licensePom.text
@@ -351,9 +358,8 @@ public class TilesMavenLifecycleParticipantTest {
 			}
 
 			@Override
-			protected Artifact resolveTile(MavenSession mavenSession, Artifact tileArtifact) throws MavenExecutionException {
+			protected Artifact resolveTile(MavenSession mavenSession, MavenProject project, Artifact tileArtifact) throws MavenExecutionException {
 				tileArtifact.file = new File("src/test/resources/${tileArtifact.artifactId}.xml")
-
 				return tileArtifact
 			}
 		}

